### DISCUSSION TITLE
duration for OutputEvents table

### DISCUSSION THREAD

**Discussion by sunukp** [MIT-LCP/mimic-code#1201] [MIMIC-General]
The table InputEvents has columns StartTime & EndTime so I can easily compute the event duration.
However OutputEvents has only one timestamp (ChartTime), so there is not an obvious way to obtain duration. Is there a guideline for this? I am looking at CSF drainage in particular. Since this seems to be recorded hourly, is it ok to say that the drainage occurred over the past hour? There are 5~20hour gaps in the data at times, which is the reason I am asking.

**Comment by alistairewj** [comment]
I usually consider *outputevents* data to be documented at an end time, and for something continual like what you mentioned (CSF drainage), then I also assume it's an hourly value. Unfortunately those 5 hour gaps could be either the drainage over 5 hours, or they simply started it again on the hour and it corresponds to < 1 hour of volume. There isn't a consistent way to tell. for the urine output queries, I wrote some approaches to basically say "if this has been going on for at least 3 hours, calculate the rate", and you may find those useful/adaptable to the CSF data.

**Reply by alistairewj** [reply]
I should also mention that the front-end interface used for documentation lets you specify the cadence of data display, and it's usually set to 1 hour, and that's why most of the time the documentation corresponds to the previous hour.
